activerecord/lib/arel/alias_predication.rb

```diff
@@ -3,7 +3,7 @@
 module Arel # :nodoc: all
   module AliasPredication
     def as(other)
-      Nodes::As.new self, Nodes::SqlLiteral.new(other)
+      Nodes::As.new self, Nodes::SqlLiteral.new(other, retryable: true)
```

Comment:
I think this technically would need to be quoted to be strictly safe, though maybe it's fine because injecting an expression here would be a clear abuse of the API (and this isn't a security check, it's an application-sanity one).